### PR TITLE
watcher/src - cosmwasm watcher

### PR DESCRIPTION
**Pull request by panoel**

### CODE REVIEW

File: watcher/src/watchers/CosmwasmWatcher.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+   * Taken from https://github.com/terra-money/terra.js/blob/9e5f553de3ff3e975eaaf91b1f06e45658b1a5e0/src/util/hash.ts
+   */
+  hexToHash(data: string): string {
+    return SHA256.hash(Base64.parse(data)).toString().toUpperCase();
+  }
+
+  async getFinalizedBlockNumber(): Promise<number> {
+    const result = (await axios.get(`${this.rpc}/${this.latestBlockTag}`)).data;
+    if (result && result.block.header.height) {
+      var blockHeight = result.block.header.height;
```
Comment (heyitaki) [inline watcher/src/watchers/CosmwasmWatcher.ts:45]: Nit: can we replace uses of `var` with `let`/`const`?

**Review by heyitaki — CHANGES_REQUESTED** [review]

File: watcher/src/watchers/CosmwasmWatcher.ts (hunk 1 above)
Comment (panoel) [reply watcher/src/watchers/CosmwasmWatcher.ts:45]: Oops.  Fixed.

File: watcher/src/watchers/CosmwasmWatcher.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        var vaaKey: string = '';
+        var numTxs = 0;
+        if (blockResult.block.data.txs) {
+          numTxs = blockResult.block.data.txs.length;
+        }
+        for (var i = 0; i < numTxs; i++) {
+          var hash = this.hexToHash(blockResult.block.data.txs[i]);
+          this.logger.info('blockNumber = ' + blockNumber + ', txHash[' + i + '] = ' + hash);
+          const hashResult = (await axios.get(`${this.rpc}/${this.hashTag}${hash}`)).data;
+          if (hashResult && hashResult.tx_response.events) {
```
[the remark this answers is not in the document]
Comment (panoel) [reply watcher/src/watchers/CosmwasmWatcher.ts:83]: Added more types.

[the remark this answers is not in the document]
Comment (panoel) [reply watcher/src/watchers/CosmwasmWatcher.ts:91]: Added.

**Review by heyitaki — APPROVED** [review]

**Pull request merged**